### PR TITLE
chore(f3): upgrade go-f3 to v0.7.0

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- upgrade `go-f3` to v0.7.0
- bump Go dependencies

CI logs: https://github.com/ChainSafe/forest/actions/runs/11224280536/job/31201086427?pr=4853#step:8:204
```
  2024-10-07T21:42:41.042Z	INFO	f3	go-f3@v0.7.0/host.go:880	reached a decision	{"instance": 2, "ecHeadEpoch": 13}
  2024-10-07T21:42:41.051Z	INFO	f3	go-f3@v0.7.0/host.go:410	backing off for: 29.52s
  [202](https://github.com/ChainSafe/forest/actions/runs/11224280536/job/31201086427?pr=4853#step:8:204)4-10-07T21:42:44.045Z	INFO	f3	go-f3@v0.7.0/host.go:880	reached a decision	{"instance": 3, "ecHeadEpoch": 18}
  2024-10-07T21:42:49.037Z	INFO	f3	go-f3@v0.7.0/host.go:880	reached a decision	{"instance": 4, "ecHeadEpoch": 19}
  2024-10-07T21:43:05.071Z	INFO	f3	go-f3@v0.7.0/host.go:880	reached a decision	{"instance": 5, "ecHeadEpoch": 19}
  2024-10-07T21:43:05.090Z	INFO	f3	go-f3@v0.7.0/host.go:410	backing off for: 16s
  2024-10-07T21:43:20.056Z	INFO	f3	go-f3@v0.7.0/host.go:880	reached a decision	{"instance": 6, "ecHeadEpoch": 19}
  2024-10-07T21:43:20.066Z	INFO	f3	go-f3@v0.7.0/host.go:410	backing off for: 29.52s
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: f3-sidecar/run.go [hunk 1]
```diff
@@ -84,15 +86,37 @@ func run(ctx context.Context, rpcEndpoint string, f3RpcEndpoint string, initialP
...
-		manifestProvider = manifest.NewStaticManifestProvider(m)
+		if manifestProvider, err = manifest.NewStaticManifestProvider(m); err != nil {
+			return err
+		}
 	default:
 		logger.Infof("Using dynamic manifest provider at %s", manifestServerID)
 		manifestDS := namespace.Wrap(ds, datastore.NewKey("/f3-dynamic-manifest"))
-		manifestProvider = manifest.NewDynamicManifestProvider(m, manifestDS, p2p.PubSub, manifestServerID)
+		primaryNetworkName := m.NetworkName
+		filter := func(m *manifest.Manifest) error {
```
Comment (hanabi1224) [inline f3-sidecar/run.go:96]: The filter logic is taken from Lotus

File: f3-sidecar/.gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 # The default f3 database folder
 /f3-db
+# The binary
```
Comment (LesnyRumcajs) [inline f3-sidecar/.gitignore:3]: Would it make sense to produce the binary in a dedicated directory, e.g., `out/`?

File: src/daemon/mod.rs [hunk 3]
```diff
@@ -443,8 +440,8 @@ pub(super) async fn start(
                     crate::rpc::f3::get_f3_rpc_endpoint().to_string(),
                     initial_power_table,
                     finality,
-                    std::env::var("FOREST_F3_DB_PATH")
-                        .unwrap_or(default_f3_db_path.display().to_string()),
+                    std::env::var("FOREST_F3_ROOT")
+                        .unwrap_or(default_f3_root.display().to_string()),
```
Comment (LesnyRumcajs) [inline src/daemon/mod.rs:444]: Is there any documentation we should update? If not, is there any documentation we should create? :D

**Review by LesnyRumcajs — APPROVED** [review]

File: src/daemon/mod.rs (hunk 3 above)
Comment (hanabi1224) [reply src/daemon/mod.rs:444]: Will make a follow-up PR in the docs repo

File: f3-sidecar/.gitignore (hunk 2 above)
Comment (hanabi1224) [reply f3-sidecar/.gitignore:3]: We don't generate the binary by default. The path is the default output of go build . This PR is Adding it to gitignore just for convenience

**Review by ruseinov — APPROVED** [review]

**Pull request merged**